src/main/webapp/app/course/plagiarism-cases/student-view/detail-view/plagiarism-case-student-detail-view.component.ts

```diff
@@ -32,18 +32,34 @@ export class PlagiarismCaseStudentDetailViewComponent implements OnInit, OnDestr
     private postsSubscription: Subscription;
     posts: Post[];
 
+    affectedExerciseRouterLink: (string | number)[];
+
     constructor(protected metisService: MetisService, private plagiarismCasesService: PlagiarismCasesService, private activatedRoute: ActivatedRoute) {}
 
     ngOnInit(): void {
-        this.paramSubscription = combineLatest({ params: this.activatedRoute.parent!.parent!.params }).subscribe((routeParams: { params: Params }) => {
-            const { params } = routeParams;
-            this.courseId = params.courseId;
-            this.plagiarismCaseId = Number(this.activatedRoute.snapshot.paramMap.get('plagiarismCaseId'));
+        this.paramSubscription = combineLatest({
+            ancestorParams: this.activatedRoute.parent!.parent!.params,
+            params: this.activatedRoute.params,
+        }).subscribe(({ ancestorParams, params }: { ancestorParams: Params; params: Params }) => {
+            this.courseId = ancestorParams.courseId;
+            this.plagiarismCaseId = Number(params.plagiarismCaseId);
+            if (this.plagiarismCase?.id === this.plagiarismCaseId) {
```

Comment:
❓Question: Just to confirm, this is if the plagiarismCase is already loaded right? How could this happen that it is loaded before the ngOnInit?

Reply:
Yes, you are right. The aim is fixing the issue where the plagiarism case is not updated when a notification was clicked (or when the route changes without a page load for any other potential reason). It also works as expected for the initial load because before any `plagiarismCase` is loaded, the `if` condition becomes `undefined === (someNumber)` so it still sends the request. The client test added in https://github.com/ls1intum/Artemis/pull/5620/commits/a30a65d80b7d26a42a536a4df42a59310f53463d and the test just above the new one that is partially modified may explain the purpose better along with the commit message.